### PR TITLE
JSON serialization improvements

### PR DESCRIPTION
**Pull request by gewarren**
- Adds article on required properties - [preview link](https://review.learn.microsoft.com/en-us/dotnet/standard/serialization/system-text-json/required-properties?branch=pr-en-us-31335)
- Moves VB support section to a separate file
- Improves introductions to provide more background info
- Removes some outdated links

Contributes to #30484.

### CODE REVIEW

File: docs/standard/serialization/system-text-json/overview.md [hunk 1]
```diff
@@ -68,6 +68,7 @@ The `System.Text.Json` types are thread-safe, including:
 * [Enable case-insensitive matching](character-casing.md)
 * [Customize property names and values](customize-properties.md)
 * [Ignore properties](ignore-properties.md)
+* [Require properties](required-properties.md)
```
Comment (eiriktsarpalis) [inline docs/standard/serialization/system-text-json/overview.md:71]:
```suggestion
* [Required properties](required-properties.md)
```

**Review by eiriktsarpalis — APPROVED** [review]
Thanks

File: docs/standard/serialization/system-text-json/source-generation-modes.md [hunk 2]
```diff
@@ -65,22 +65,22 @@ By default, `JsonSerializer` collects metadata at run time by using [reflection]
...
 
 In both reflection and source generation modes:
 
-* Only public properties and public fields are supported.
-* Only public constructors can be used for deserialization.
+* Only `public` properties and `public` fields are supported<sup>\*</sup>.
+* Only `public` constructors can be used for deserialization.
 
-For information about the outstanding request to add support for non-public members, see GitHub issue [dotnet/runtime#31511](https://github.com/dotnet/runtime/issues/31511). Even if that request is implemented, source generation mode will still be limited to support for public members.
+<sup>\*</sup>Starting in .NET 7, you can use custom JSON contracts to include `private` properties and fields in serialization.
```
Comment (eiriktsarpalis) [inline docs/standard/serialization/system-text-json/source-generation-modes.md:83]: Might want to link to the contract customization documentation here once it has been added.

Comment (gewarren) [reply docs/standard/serialization/system-text-json/source-generation-modes.md:83]: Yes, I plan on doing that. Thanks for the suggestion!

**Pull request merged**